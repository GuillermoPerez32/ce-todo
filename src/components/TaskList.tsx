import clsx from "clsx";
import { useTaskStore } from "../services/task";
import { renderFormattedText } from "../utils/renderFormattedText";

export const TaskList = () => {
  const tasks = useTaskStore((state) => state.tasks);
  const setTaskDone = useTaskStore((state) => state.setDone);
  const removeTask = useTaskStore((state) => state.remove);

  return (
    <div className="mt-6">
      {tasks.map((task) => (
        <div key={task.id} className="p-2 mt-1 flex items-center">
          <input
            type="checkbox"
            className="focus:ring-blue-500 h-4 w-4"
            checked={task.done}
            onChange={(e) => {
              setTaskDone(task.id, e.target.checked);
              setTimeout(() => {
                removeTask(task.id);
              }, 500);
            }}
          />
          <p
            className={clsx("ml-2 transition-opacity duration-500", {
              "line-through opacity-0": task.done,
            })}
          >
            {renderFormattedText(task.content)}
          </p>
        </div>
      ))}
    </div>
  );
};
