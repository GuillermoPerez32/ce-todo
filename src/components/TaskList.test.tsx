import { test, describe, expect } from "vitest";
import { TaskList } from "./TaskList";
import { render } from "@testing-library/react";

describe("<TaskList />", () => {
  test("renders TaskList with label", () => {
    const { container } = render(<TaskList />);
    expect(container).toMatchSnapshot();
  });
});
