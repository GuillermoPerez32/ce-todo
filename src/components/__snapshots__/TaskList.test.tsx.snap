// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<TaskList /> > renders TaskList with label 1`] = `
<div>
  <div
    class="mt-6"
  />
</div>
`;
